centrallix/tests/t_driver.c

```diff
@@ -25,23 +25,31 @@
 /************************************************************************/
 
 
-long long test(void);
+#define RED    "\033[1m\x1B[31m"
+#define GREEN  "\033[1m\x1B[32m"
+#define RESET  "\x1B[0m"
```

Comment:
The original setup used tput (a part of the ncurses library) to determine the appropriate escape sequences for colors.  These values can change from one terminal type to another, so they should be emitted via ncurses, which uses terminfo to determine the correct escape sequences.

Reply:
(note - don't call out to /usr/bin/tput from the C code, instead use the ncurses API)

Reply:
Sounds great! One question - this means I'll need to link in ncurses with `-lncurses`. Is that something I should just add to the command in Makefile.in for compiling each .bin file...or should I handle it through the $(LIBS) variable, include something in aclocal.m4 to make sure that it's available, etc (like I think I see being done with other libraries)?

Reply:
Actually, never mind...figured out that I don't need ncurses but just some lower-level APIs that don't require extra linking.

Reply:
Sorry, I just posted and then deleted a comment about not needing to link in ncurses - I DO actually still need to do that and would still appreciate your guidance on how to do that!

Reply:
Two options: A) just put the -lncurses in the .bin Makefile.in rules and create an issue to fix it later, or B) do it "properly" now by: 1) creating an aclocal.m4 macro to properly detect ncurses (not however updating $(LIBS) as ncurses isn't needed for the server itself), 2) adjust the Makefile.in appropriately, and 3) have conditional compilation in the test driver to omit colors if the ncurses library or headers is not found (most common case: ncurses-devel or ncurses-dev not installed).

Reply:
I'm honestly fine with either way of scheduling the more complete solution.

Reply:
(the aclocal.m4 macro should be called from configure.ac)

Reply:
I'm working on doing ncurses detection in #16 and will request a review on that when it's ready :slightly_smiling_face: Thanks for the advice!